core/crypto/vrf/vrf_provider_impl.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/vrf/vrf_provider_impl.hpp"
+
+#include <gsl/span>
+
+namespace kagome::crypto {
+
+  VRFProviderImpl::VRFProviderImpl(std::shared_ptr<CSPRNG> generator)
+      : generator_{std::move(generator)} {}
+
+  VRFKeypair VRFProviderImpl::generateKeypair() const {
+    auto seed = generator_->randomBytes(SR25519_SEED_SIZE);
+
+    std::vector<uint8_t> kp(SR25519_KEYPAIR_SIZE, 0);
+    sr25519_keypair_from_seed(kp.data(), seed.data());
+
+    VRFKeypair result{};
+    std::copy(kp.begin(), kp.begin() + SR25519_SECRET_SIZE,
+              result.secret_key.begin());
+    std::copy(kp.begin() + SR25519_SECRET_SIZE,
+              kp.begin() + SR25519_SECRET_SIZE + SR25519_PUBLIC_SIZE,
+              result.public_key.begin());
+
+    return result;
+  }
+
+  std::array<uint8_t, 32> uint256_t_to_bytes(
+      boost::multiprecision::uint256_t const &i) {
+    auto count = i.backend().size();
+    auto tsize = sizeof(boost::multiprecision::limb_type);
+    auto copy_count = count * tsize;
+
+    std::array<uint8_t, 32> output{};
+    memcpy(output.data(), i.backend().limbs(), copy_count);
```

Comment:
std::copy

Reply:
limbs is c-style array. When `std::copy(limbs, limbs + copy_count, output.begin());` is applied pointer arithmetic is used, which is prohibited by clang-tidy